apache/incubator-metron
# Metron-383 Create Ambari Service Definition for Metron Parsers

More detailed instructions for testing are coming soon, but I wanted to get community eyes on this asap.

This PR addresses [https://issues.apache.org/jira/browse/METRON-383](https://issues.apache.org/jira/browse/METRON-383)

To test this PR:
- You need to build the rpm's and make them available in a "/localrepo" directory on the node you plan to use as Master for the parsers.
- Copy or mount the services and stacks content into the correct ambari-server resources directories. For example
  - incubator-metron/metron-deployment/packaging/ambari/src/main/resources/stacks/HDP/2.3/services/PARSERS -> /var/lib/ambari-server/resources/stacks/HDP/2.3/services/PARSERS
  - incubator-metron/metron-deployment/packaging/ambari/src/main/resources/common-services/PARSERS -> /var/lib/ambari-server/resources/common-services/PARSERS
- Use Ambari 2.4
- Install on a node with a Kafka Broker and ZK client (service advisor recommendations to come in a separate PR)
- You can check both a full install as well as an install of HDFS, Kakfa, Zookeeper, and Storm followed after by Metron Parsers.

Cleanup/delete service
- delete the service via Ambari UI or REST API
Run the following commands from the master node, swapping out the zookeeper quorum as appropriate:

```
#!/bin/bash

# kafka
# (in your kafka settings, first enable delete.topic.enable=true and restart Kafka)
for topic in bro parser_error parser_invalid snort yaf
do
    /usr/hdp/current/kafka-broker/bin/kafka-topics.sh \
            --zookeeper amb1.service.consul:2181,amb2.service.consul:2181,amb3.service.consul:2181 \
            --delete \
            --topic $topic
done

# storm kill
for i in bro yaf snort
do
    storm kill $i
done

# yum delete
yum -y erase metron-parsers metron-common

# final file delete
rm -rf /usr/metron

# zookeeper
/usr/hdp/current/zookeeper-client/bin/zkCli.sh -server 127.0.0.1:2181

# once connected to the client do this:
rmr /metron
```
